src/Workspaces/Core/Portable/Shared/Extensions/ObjectExtensions.TypeSwitch.cs

```diff
@@ -7877,6 +7877,228 @@ internal static partial class ObjectExtensions
+           where TDerivedType1 : TBaseType
+           where TDerivedType2 : TBaseType
+           where TDerivedType3 : TBaseType
+           where TDerivedType4 : TBaseType
+           where TDerivedType5 : TBaseType
+           where TDerivedType6 : TBaseType
+           where TDerivedType7 : TBaseType
+           where TDerivedType8 : TBaseType
+           where TDerivedType9 : TBaseType
+           where TDerivedType10 : TBaseType
+           where TDerivedType11 : TBaseType
+           where TDerivedType12 : TBaseType
+           where TDerivedType13 : TBaseType
+           where TDerivedType14 : TBaseType
+           where TDerivedType15 : TBaseType
+           where TDerivedType16 : TBaseType
+           where TDerivedType17 : TBaseType
+           where TDerivedType18 : TBaseType
+           where TDerivedType19 : TBaseType
+           where TDerivedType20 : TBaseType
+           where TDerivedType21 : TBaseType
+           where TDerivedType22 : TBaseType
+           where TDerivedType23 : TBaseType
+           where TDerivedType24 : TBaseType
+           where TDerivedType25 : TBaseType
+           where TDerivedType26 : TBaseType
+           where TDerivedType27 : TBaseType
+           where TDerivedType28 : TBaseType
+           where TDerivedType29 : TBaseType
+           where TDerivedType30 : TBaseType
+           where TDerivedType31 : TBaseType
+           where TDerivedType32 : TBaseType
+           where TDerivedType33 : TBaseType
+           where TDerivedType34 : TBaseType
+           where TDerivedType35 : TBaseType
+           where TDerivedType36 : TBaseType
+           where TDerivedType37 : TBaseType
+           where TDerivedType38 : TBaseType
+           where TDerivedType39 : TBaseType
+           where TDerivedType40 : TBaseType
+           where TDerivedType41 : TBaseType
+           where TDerivedType42 : TBaseType
```

Comment:
Yeah, ugly, and predicates on a NodeType that isn't there. Rather than implement that, you could just go implement type matching in the compiler. :smile: